persistence/src/main/scala/za/co/absa/spline/persistence/ArangoManager.scala

```diff
@@ -127,18 +128,21 @@ class ArangoManagerImpl(
       _ <- if (exists && !dropIfExists)
         throw new IllegalArgumentException(s"Arango Database ${db.name()} already exists")
       else if (exists && dropIfExists) {
-        log.debug("Drop existing database")
+        log.info(s"Drop database: ${db.name}")
         db.drop().toScala
       }
-      else Future.successful(Unit)
-    } yield Unit
+      else Future.successful({})
+    } yield {}
```

Comment:
In my opinion, `{}` instead of `Unit` is making the code harder to undertand.

Reply:
The reason why I removed `Unit` is because it is _not_ an instance of the type `Unit`. The only instance of it is `()`. So the proper writing of the end of that `for` block would be one of these: `yield ()`, `yield {()}` or `yield {}`

Reply:
OK, so it's correct. Still, it's a bit confusing. There is a `Future.unit` but only in Scala 2.12:
https://www.scala-lang.org/api/2.12.15/scala/concurrent/Future$.html#unit:scala.concurrent.Future[Unit]